expertsystem/io/dot.py

```diff
@@ -30,55 +32,88 @@ def write(instance: object, filename: str) -> None:
         stream.write(output_str)
 
 
-_DOT_HEAD = """
-digraph {
+_DOT_HEAD = """digraph {
     rankdir=LR;
-    node [shape=point];
-    edge [arrowhead=none, labelfloat=true];
-    """
+    node [shape=point, width=0];
+    edge [arrowhead=none];
+"""
 _DOT_TAIL = "}\n"
 _DOT_RANK_SAME = "    {{ rank=same {} }};\n"
 _DOT_DEFAULT_NODE = '    "{}" [shape=none, label="{}"];\n'
 _DOT_DEFAULT_EDGE = '    "{}" -> "{}";\n'
 _DOT_LABEL_EDGE = '    "{}" -> "{}" [label="{}"];\n'
 
 
-def __graph_to_dot(graph: StateTransitionGraph) -> str:
-    def node_name(edge_id: int, node_id: Optional[int] = None) -> str:
-        if node_id is None:
-            return f"edge{edge_id}"
-        return f"node{node_id}"
+def __graph_list_to_dot(graphs: List[StateTransitionGraph]) -> str:
+    dot_source = _DOT_HEAD
+    for i, graph in enumerate(graphs):
+        dot_source += __graph_to_dot_content(graph, prefix=f"g{i}_")
+    dot_source += _DOT_TAIL
+    return dot_source
 
-    def format_particle(node_edge_ids: List[int]) -> str:
-        name_list = [f'"{node_name(i)}"' for i in node_edge_ids]
-        return ",".join(name_list)
 
+def __graph_to_dot(graph: StateTransitionGraph) -> str:
     dot_source = _DOT_HEAD
+    dot_source += __graph_to_dot_content(graph)
+    dot_source += _DOT_TAIL
+    return dot_source
 
+
+def __graph_to_dot_content(
```

Comment:
Proposal: change this back to `__graph_to_dot`, but include the prefix part. Remove `__graph_to_dot`

Reply:
I have to think about this. Probably, the construction of wrapping the 'content' into `_DOT_HEAD` and a `_DOT_TAIL` can be better done with a decorator.

Reply:
For the decorator, see 82ade244b2b8ab8031c870e936a5591dbc212faf
I'm not sure whether combining functions here makes much sense. How would you do it?

Reply:
Well i meant, take this function rename it `__graph_to_dot` and also add the header and footer back in.

Reply:
I guess I misunderstood this DOT syntax then